core/unit_test/TestGraph.hpp

```diff
@@ -103,28 +103,30 @@ struct TEST_CATEGORY_FIXTURE(graph) : public ::testing::Test {
   }
 };
 
+// If the counts are correct, all nodes have successfully executed
+// their kernel. However, it does not necessarily imply that dependencies
+// were observed.
+#define ASSERT_THAT_ALL_NODES_EXECUTED(exec, expt_count, expt_bugs) \
+  Kokkos::deep_copy(exec, count_host, count);                       \
+  Kokkos::deep_copy(exec, bugs_host, bugs);                         \
+  exec.fence();                                                     \
+  ASSERT_EQ(expt_count, count_host());                              \
+  ASSERT_EQ(expt_bugs, bugs_host());
+
 TEST_F(TEST_CATEGORY_FIXTURE(graph), submit_once) {
   auto graph =
       Kokkos::Experimental::create_graph<TEST_EXECSPACE>([&](auto root) {
         root.then_parallel_for(1, count_functor{count, bugs, 0, 0});
       });
   graph.submit();
-  Kokkos::deep_copy(graph.get_execution_space(), count_host, count);
-  Kokkos::deep_copy(graph.get_execution_space(), bugs_host, bugs);
-  graph.get_execution_space().fence();
-  ASSERT_EQ(1, count_host());
-  ASSERT_EQ(0, bugs_host());
```

Comment:
I am curious what other reviewers think.  I am not 100% sold that it helps with readability.

Reply:
OK. I got your point.

I changed to creating a custom "check that rank-0 view contains a given value" assertion. So it reduces the number of "repetitive" lines and to me, it also makes things easier to read :mag: